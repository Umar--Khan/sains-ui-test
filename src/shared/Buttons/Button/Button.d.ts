import { MouseEventHandler } from 'react';

export interface ButtonProps {
  ariaLabel?: string;
  className?: string;
  color?: string;
  id?: string;
  onClick?: MouseEventHandler<HTMLButtonElement>;
}
