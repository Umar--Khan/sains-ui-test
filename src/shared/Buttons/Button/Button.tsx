import { PropsWithChildren } from 'react';
import styled from 'styled-components';

import * as colors from 'src/styles/_colors.module.scss';

import { MouseEventHandler } from 'react';

export interface ButtonProps {
  ariaLabel?: string;
  className?: string;
  id?: string;
  onClick?: MouseEventHandler<HTMLButtonElement>;
}

const StyledButton = styled.button`
  background-color: ${colors.orange};
  color: ${colors.white};
  width: 100%;
  font-size: 1.25rem;
  font-weight: 700;
  box-shadow: 0 0 0 5px transparent;
  border-radius: 5px;
  padding: 0.5rem 1rem;
  line-height: 1.5rem;
  border: 0;
  cursor: pointer;
`;

const Button = ({
  children,
  className,
  id,
  onClick,
  ariaLabel,
  ...rest
}: PropsWithChildren<ButtonProps>) => (
  <StyledButton
    className={className}
    id={id}
    onClick={onClick}
    aria-label={ariaLabel}
    {...rest}
  >
    {children}
  </StyledButton>
);

export default Button;
