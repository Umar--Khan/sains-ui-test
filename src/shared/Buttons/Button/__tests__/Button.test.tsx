import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';

import Button, { ButtonProps } from '../Button';

describe('<Button />', () => {
  it('Should render correctly', () => {
    const props = {
      name: 'button-name',
      className: 'button-class-name',
      ariaLabel: 'button-aria-label',
      id: 'button-id',
    } as ButtonProps;

    const component = renderer.create(<Button {...props}>Button</Button>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Should call the onClick prop', () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>Button</Button>);
    fireEvent.click(screen.getByText('Button'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
