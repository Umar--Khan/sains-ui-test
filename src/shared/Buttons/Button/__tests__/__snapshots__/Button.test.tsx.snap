// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render correctly 1`] = `
<button
  aria-label="button-aria-label"
  className="sc-bdfBwQ ibNfPE button-class-name"
  id="button-id"
  name="button-name"
>
  Button
</button>
`;
